[core] handle unserializable user exception (#44878)
ray-project/ray: doc/source/ray-core/fault_tolerance/tasks.rst

  :start-after: __task_exceptions_begin__
  :end-before: __task_exceptions_end__

> If the user's exception cannot be serialized, it would be converted to a ``RayError``:

Reviewer: ```suggestion
If Ray can't serialize the user's exception, it converts the exception to a ``RayError``.
```

Reviewer: Agree with @jjyao's rewrite.

.. literalinclude:: ../doc_code/task_exceptions.py
  :language: python
  :start-after: __unserializable_exceptions_begin__
  :end-before: __unserializable_exceptions_end__

> 

Reviewer 2: Move it down to after line 49. Currently its in the middle of subclass section.

  :start-after: __task_exceptions_begin__
  :end-before: __task_exceptions_end__

> If the user's exception cannot be serialized, it would be converted to a RayError:

Reviewer 3: not sure why it ends with colon `:`.

and also maybe change to active tense, like:

````
If Ray cannot serialize the user's exception, it will convert the exception to a ``RayError``

````

Author: Thanks! Missed some sample code before. Fixed
